mitodl/ocw-studio
# Refactor Youtube token generation

#### Pre-Flight checklist



#### What are the relevant tickets?
Closes #1458 

#### What's this PR do?
Uses a web-based non-OOB flow method to generate Youtube tokens.

#### How should this be manually tested?
- You will need to use ngrok or similar for this to work locally:  `ngrok http 8043`.
- Follow the updated README instructions for YouTube integration, using the https ngrok domain for URI values.
- Set `OCW_STUDIO_ENVIRONMENT=devdev` to bypass use of fake minio s3 buckets.  Set `AWS_` .env values that of RC.  Set `OCW_STUDIO_BASE_URL` to the ngrok secure url.
- Upload a small video to a site's videos_final folder, then sync.
- Check that there is a new `Video` object via django admin.
- Using postman or curl, send the following request to http://localhost:8043/api/transcode-jobs/:

```
{
  "version": "0",
  "id": "c120fe11-87db-c292-b3e5-1cc90740f6e1",
  "detail-type": "MediaConvert Job State Change",
  "source": "aws.mediaconvert",
  "account": "<AWS_ACCOUNT_ID>",
  "time": "2021-08-05T16:52:33Z",
  "region": "us-east-1",
  "resources": [
    "arn:aws:mediaconvert:us-east-1:11111111:jobs/111111111-abplcg"
  ],
  "detail": {
    "timestamp": 1628172900136,
    "accountId": "<AWS_ACCOUNT_ID>",
    "queue": "arn:aws:mediaconvert:us-east-1:<AWS_ACCOUNT_ID>:queues/Default",
    "jobId": "<VIDEOJOB_ID>",
    "status": "COMPLETE",
    "userMetadata": {},
    "outputGroupDetails": [
      {
        "outputDetails": [
          {
            "outputFilePaths": [
              "s3://ol-ocw-studio-app-qa/aws_mediaconvert_transcodes/<VIDEO_BASE_NAME>_youtube.mp4"
            ],
            "durationInMs": 132033,
            "videoDetails": {
              "widthInPx": 1280,
              "heightInPx": 720
            }
          }
        ],
        "type": "FILE_GROUP"
      }
    ]
  }
}
```

where `VIDEO_BASE_NAME` is derived from the `Video.source_key` value, for example:
`gdrive_uploads/course1/14EvKgqg1eQqzxoo6YxvmxOxgf-HPn0mM/medium.mp4`. -> `course1/14EvKgqg1eQqzxoo6YxvmxOxgf-HPn0mM/medium`
- Wait a few minutes and verify that the video is uploaded to Youtube, processed, and is playable (refresh the Django admin page for the video every few minutes).
